### PR TITLE
Markdown doc for `ErrorOutOfGasMemoryCopy` state

### PR DESCRIPTION
**Pull request by silathdiir**
Circuit PR https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/1288

### CODE REVIEW

File: specs/error_state/ErrorOutOfGasMemoryCopy.md
Comment (KimiWu123) [inline specs/error_state/ErrorOutOfGasMemoryCopy.md:80]: Add this TODO in the end of this file


### Code 
```
TODO: add after circuit merge first.
```

**Review by KimiWu123 — APPROVED** [review]
LGTM, only one minor comment.

File: specs/error_state/ErrorOutOfGasMemoryCopy.md
Comment (silathdiir) [reply specs/error_state/ErrorOutOfGasMemoryCopy.md:80]: Add this TODO in commit [0252dc9](https://github.com/privacy-scaling-explorations/zkevm-specs/pull/397/commits/0252dc9ec2af3004f666449f3bdd10f5f94e00d3).

Comment (KimiWu123) [reply specs/error_state/ErrorOutOfGasMemoryCopy.md:80]: sorry, I might not make it clear. The `Code` section also needs to be added, otherwise no one knows what should be added after circuit merged.

截圖 2023-04-14 下午4 46 51

Comment (silathdiir) [reply specs/error_state/ErrorOutOfGasMemoryCopy.md:80]: Sorry. Add the `Code` section in [e053557](https://github.com/privacy-scaling-explorations/zkevm-specs/pull/397/commits/e05355775f237c6acaf06d3f27960ff431062848).

File: specs/error_state/ErrorOutOfGasMemoryCopy.md
Comment (CPerezz) [inline specs/error_state/ErrorOutOfGasMemoryCopy.md:84]: Pleasem let's also file an issue for that to keep track of it.

**Review by CPerezz — APPROVED** [review]
LGTM modulo the issue creation.

File: specs/error_state/ErrorOutOfGasMemoryCopy.md
Comment (CPerezz) [inline specs/error_state/ErrorOutOfGasMemoryCopy.md:61]: This is missing one constraint:
```
Current opcode is one of:  `CALLDATACOPY`, `CODECOPY`, `EXTCODECOPY` and `RETURNDATACOPY`
```

**Review by CPerezz — CHANGES_REQUESTED** [review]
Sorry, saw that we were lacking constraints late.

File: specs/error_state/ErrorOutOfGasMemoryCopy.md
Comment (silathdiir) [reply specs/error_state/ErrorOutOfGasMemoryCopy.md]: Add this constraint as the first in commit [e0fc661](https://github.com/privacy-scaling-explorations/zkevm-specs/pull/397/commits/e0fc661280ba78a723384645206ca93334f18e91).

File: specs/error_state/ErrorOutOfGasMemoryCopy.md
Comment (silathdiir) [reply specs/error_state/ErrorOutOfGasMemoryCopy.md:85]: Add a new issue for spec code https://github.com/privacy-scaling-explorations/zkevm-specs/issues/414.

**Review by CPerezz — APPROVED** [review]
LGTM

**Pull request merged**
